api/src/main/java/com/marketplace/controller/CategoryController.java

```diff
@@ -0,0 +1,46 @@
+package com.marketplace.controller;
+
+import com.marketplace.entity.Category;
+import com.marketplace.service.impl.CategoryService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.*;
+import java.util.List;
+
+@RestController
+@RequestMapping("/categories")
+public class CategoryController {
+
+    @Autowired
```

Comment:
Sur les nouvelles versions de Spring, on utilise le constructor injection à la place de @Autowired. Ca permet d'avoir l'immutabilité et une meilleure gestion des injections.
Ca ressemble à ça :
private final CategoryService categoryService;

puis soit avant ta class  tu utilises l'annotation Lombok @RequiredArgsConstructor (préférable dans notre cas) soit tu déclares ton constructor comme ça
public CategoryController(CategoryService categoryService) { this.categoryService = categoryService; }